src/components/Company/DocumentSigner/SignatureForm/SignatureForm.test.tsx

```diff
@@ -0,0 +1,92 @@
+import { beforeEach, describe, expect, it, vi } from 'vitest'
+import { render, screen, waitFor } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { HttpResponse } from 'msw'
+import { SignatureForm } from './SignatureForm'
+import { GustoTestApiProvider } from '@/test/GustoTestApiProvider'
+import { setupApiTestMocks } from '@/test/mocks/apiServer'
+import { companyEvents } from '@/shared/constants'
+import { server } from '@/test/mocks/server'
+import { handleSignCompanyForm } from '@/test/mocks/apis/company_forms'
+
+vi.mock('@/hooks/useContainerBreakpoints/useContainerBreakpoints', async () => {
+  const actual = await vi.importActual('@/hooks/useContainerBreakpoints/useContainerBreakpoints')
+  return {
+    ...actual,
+    default: () => ['base', 'small', 'medium'],
+    useContainerBreakpoints: () => ['base', 'small', 'medium'],
+  }
+})
+
+const testForm = {
+  uuid: 'form-123',
+  name: 'Test Form',
+  status: 'not_signed',
+  form_type: 'company',
+  created_at: '2024-05-29T12:00:00Z',
+  updated_at: '2024-05-29T12:30:00Z',
+  requires_signing: true,
+}
+
+describe('SignatureForm', () => {
+  const mockOnEvent = vi.fn()
+
+  beforeEach(() => {
+    setupApiTestMocks()
+    mockOnEvent.mockClear()
+    vi.resetModules()
+
+    server.use(handleSignCompanyForm(() => HttpResponse.json(testForm)))
+  })
+
+  it('fires the correct events when form is submitted successfully', async () => {
+    const user = userEvent.setup()
+
+    render(
+      <GustoTestApiProvider>
+        <SignatureForm form={testForm} companyId="company-123" onEvent={mockOnEvent} />
+      </GustoTestApiProvider>,
+    )
+
+    await waitFor(() => {
+      expect(screen.getByText(/Signature required for/)).toBeInTheDocument()
+    })
+
+    await user.type(screen.getByLabelText('Signature'), 'John Doe')
+
+    const checkbox = screen.getAllByLabelText(
+      'I am the signatory and I agree to sign electronically',
+    )[0]
```

Comment:
Are there two labels with the same text? That could be an a11y issue but i'm not sure. Just guessing because getByLabel uses a role/element to do so.

Reply:
Yeah this is our lingering single input checkbox group issue that keeps surfacing. both the group and the checkbox share the same label. In a world where we were off react aria, we could have our own single checkbox validation and this would go away. Not sure what the best workaround is in the meantime.

Reply:
One quick stopgap would be to provide a distinct label for the group from the label on the checkbox input itself. maybe we can chat about it when you get in